### PR TITLE
REACH-147 Node view for List.Create

### PR DESCRIPTION
**Pull request by pbidenko**
Created node view for List.Create.

### CODE REVIEW

File: app/app.html [hunk 1]
```diff
@@ -420,6 +420,16 @@
 
 </script>
 
+<script type="text/template" id="node-list-template">
```
Comment (pbidenko) [inline app/app.html]: New template.

File: app/scripts/helpers/CommandsMap.js [hunk 2]
```diff
@@ -8,6 +8,7 @@ define( ['CreateNodeCommand', 'CreateProxyNodeCommand', 'DeleteModelCommand', 'M
             removeConnection: MakeConnectionCommand,
             addConnection: MakeConnectionCommand,
             run: RunCancelCommand,
-            addWorkspace: CreateCustomNodeCommand
+            addWorkspace: CreateCustomNodeCommand,
+            modelEvent: ModelEventCommand
```
Comment (pbidenko) [inline app/scripts/helpers/CommandsMap.js]: Added link to ModelEventCommand

File: app/scripts/models/nodes/ListNode.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+define(['Node', 'FLOOD'], function (Node, FLOOD) {
```
Comment (pbidenko) [inline app/scripts/models/nodes/ListNode.js]: Model for ListNode.

File: app/scripts/views/NodeViews/List.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
+define(['backbone', 'underscore', 'jquery', 'BaseNodeView'], function (Backbone, _, $, BaseNodeView) {
```
Comment (pbidenko) [inline app/scripts/views/NodeViews/List.js]: View for ListNode.

File: app/scripts/views/NodeViews/List.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+            this.model.workspace.setNodeProperty({ property: "extra", _id: this.model.get('_id'), newValue: exCopy, oldValue: ex });
+
+            this.render();
+            this.model.trigger('updateRunner');
+            this.model.workspace.run();
+        },
+
+        renderNode: function () {
+
```
Comment (pboyer) [inline app/scripts/views/NodeViews/List.js]: Ensure this doesn't get called multiple times.  NodeView does not give this guarantee:

https://github.com/pboyer/flood/blob/master/app/scripts/views/NodeViews/Base.js

File: app/scripts/views/NodeViews/List.js [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        addInput: function () {
+            var type = this.model.get('type');
+            var inputConnections = this.model.get('inputConnections');
+
+            if (type.inputs.length === 26)
+                return;
+
+            this.model.workspace.runner.post({ kind: "modelEvent", _id: this.model.get('_id'), eventName: 'AddInPort' });
+
```
Comment (pboyer) [inline app/scripts/views/NodeViews/List.js]: I created a task to expose this as an event on NodeModel:

http://adsk-oss.myjetbrains.com/youtrack/issue/REACH-169

and to remove Workspace reference from Node:
http://adsk-oss.myjetbrains.com/youtrack/issue/REACH-170

**Comment by pboyer** [comment]
Take a look at my comments above.  Otherwise, looks good.

File: app/scripts/views/NodeViews/List.js (hunk 5 above)
Comment (pbidenko) [inline app/scripts/views/NodeViews/List.js]: Use jquery's 'one' function to bind event only once.

**Comment by pbidenko** [comment]
Hi @pboyer,
I've fixed that issue with event binding. Please, take a look.

**Comment by pboyer** [comment]
Hi @pbidenko,

After this is done once, there's no reason to the find call.  Why not just set a boolean flag and only do it once?

**Comment by pbidenko** [comment]
Hi @pboyer,
I got rid off jquery find call, now using backbone events.

**Comment by pboyer** [comment]
Much better, thanks @pbidenko!

**Pull request merged**